content/en/docs/ibm/authentication.md

```diff
@@ -0,0 +1,90 @@
+1. Keep Kubernetes cluster name in a environment variable. Run command `export CLUSTER_NAME=<cluster_name>`
+1. Create a DNS domain and certificates for the IP of the service `istio-ingressgateway` in namespace `istio-system`:
+    ```SHELL
+    ibmcloud ks nlb-dns create classic --cluster $CLUSTER_NAME --ip $INGRESS_GATEWAY_IP --secret-namespace istio-system
+    ```
+1. List the registered domain names:
+    ```SHELL
+    ibmcloud ks nlb-dns ls --cluster $CLUSTER_NAME
+    ```
+1. Wait until the status of the certificate (the fourth field) of the new domain name becomes `created`. Then save the value of the column `SSL Cert Secret Name` and `Hostname` in environment variables by running these commands:
+    ```SHELL
+    export INGRESS_GATEWAY_SECRET=<the secret's name as shown in the SSL Cert Secret Name column>
+    export INGRESS_GATEWAY_HOST=<the hostname as shawn in the Hostname column>
+    ```
+    **Note**: If there are more than one entries in the output, choose the entry matches IP address from the above `LoadBalancer Ingress` of service `istio-ingressgateway`. 
+1. Create a secret named `istio-ingressgateway-certs` for the `istio-ingressgateway` pods in namespace `istio-system`:
+    ```SHELL
+    kubectl get secret $INGRESS_GATEWAY_SECRET --namespace istio-system -o yaml > istio-ingressgateway-certs.yaml
+    # In the istio-ingressgateway-certs.yaml file, change the value of name: to istio-ingressgateway-certs and save the file.
+    kubectl apply -f istio-ingressgateway-certs.yaml -n istio-system
+    kubectl rollout restart deploy istio-ingressgateway -n istio-system
+    rm istio-ingressgateway-certs.yaml
+    ```
+1. Update the gateway `kubeflow-gateway` to expose port `443`. Create resource file `kubeflow-gateway.yaml` like following by replacing `<hostname>` with the value of above `Hostname`:
+    ```YAML
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+    name: kubeflow-gateway
+    namespace: kubeflow
+    spec:
+      selector:
+        istio: ingressgateway
+    servers:
+    - hosts:
+      - '<hostname>'
+      port:
+        name: https
+        number: 443
+        protocol: HTTPS
+      tls:
+        httpsRedirect: true # sends 301 redirect for http requests
+        mode: SIMPLE
+        privateKey: /etc/istio/ingressgateway-certs/tls.key
+        serverCertificate: /etc/istio/ingressgateway-certs/tls.crt
+    ```
+1. Verify that the traffic is routed via HTTPS by using the value of above `Hostname` in your browser. It should redirect traffic from HTTP address to HTTPS address automatically.
+
+**Note**: The certificates for the NLB DNS host secret expires every **90** days. The secret in the `istio-system` namespace is automatically renewed by IBM Cloud Kubernetes Service 37 days before it expires. After this secret is updated, you must manually copy it to the secret `istio-ingressgateway-certs` by repeating commands in the step 6.
```

Comment:
Will IBM Cloud create any event/notification for the cert update? So that we can automate this process with a trigger or knative cloud event.

Reply:
Good question! I've no idea but technically it's possible but I haven't hit the first `90 days` by myself to see what it will do :)